### ISSUE TITLE
Constructor annotations not used if diamond operator used to infer constructor type arguments

### ISSUE THREAD

**Issue by patrickmeiring** [typetools/checker-framework#2183]
**Background.** When one constructs a generic type, one can let the Java compiler automatically infer the type parameters from the context by using the 'Diamond Operator' (<>).
 
**Issue.** When one uses the diamond operator in checker, the qualifiers on the type parameters are correctly inferred, but any annotations on the type itself are lost. Specifically:
```java
@MyQual List<String> myClass = new @MyQual ArrayList<String>();
```
works as expected (generates no compile error), but
```java
@MyQual List<String> myClass = new @MyQual ArrayList<>();
```
results in an assignment.type.incompatible error because the right-hand side has the default qualifier for type List, rather than `@MyQual`. This is unexpected.

I have isolated this issue to AnnotatedTypeFactory.fromNewClass(). A pull request with a fix for this issue will be provided shortly.
